accessible/workspace_treeview.component.js

```diff
@@ -0,0 +1,234 @@
+        function() { return blocklyApp.TreeView; }), blocklyApp.FieldView],
+    inputs: ['block', 'isTopBlock', 'topBlockIndex', 'level', 'parentId'],
+  })
+  .Class({
+    constructor: [blocklyApp.ClipboardService, blocklyApp.TreeService, function(_service, _service2) {
+      this.infoBlocks = {};
+      this.nextBlock = {};
+      this.sharedClipboardService = _service;
+      this.treeService = _service2;
+    }],
+    setLabelledBy: function(item,string){
+      if (!item.getAttribute('aria-labelledby')) {
+        item.setAttribute('aria-labelledby', string);
+      }
+    },
+    concatStringWithSpaces: function(){
+      var string = arguments[0];
+      for (i = 1; i < arguments.length; i++) {
+        string = string + ' ' + arguments[i];
+      }
+      return string;
+    },
+    hasNoPreviousConnection: function(block){
+      if (!block.previousConnection){
+        return 'blockly-disabled';
+      } else {
+        return undefined;
+      }
+    },
+    hasNoNextConnection: function(block){
+      if (!block.nextConnection){
+        return 'blockly-disabled';
+      } else {
+        return undefined;
+      }
+    },
+    checkParentList: function(parentList) {
+      console.log('setting parent list');
+      var tree = parentList;
+      while (tree && !tree.id.match(/^blockly-workspace-tree\d+$/)) {
+        tree = tree.parentNode;
+      }
+      if (tree && tree.getAttribute('aria-activedescendant') == parentList.id){
+        this.treeService.updateSelectedNode(parentList, tree, false);
+      }
+    },
+    addClass: function(node, classText) {
+      //ensure that node doesn't have class already in it
+      var classList = node.className;
+      classList = classList.split(' ');
+      var canAdd = true;
+      for (var i=0; i<classList.length; i++){
+        if (classList[i] == classText) {
+          canAdd = false;
+        }
+      }
+      //add class if it doesn't
+      if (canAdd) {
+        if (classList.length == 0) {
+          node.className += classText;
+        } else {
+          node.className += (' ' + classText);
+        }
+      }
+    },
+    setId: function(block){
+      if (this.isTopBlock){
+        return this.parentId+'-node0';
+      }
+      return this.treeService.createId(block);
+    },
+    getInfo: function(block) {
+      //List all inputs
+      if (this.infoBlocks[block.id]) {
+        //TODO(madeeha): is there a situation in which overwriting often unnecessarily is a problem?
+        this.infoBlocks[block.id].length = 0;
+      } else {
+        this.infoBlocks[block.id] = [];
+      }
+
+      var blockInfoList = this.infoBlocks[block.id];
+
+      for (var j = 0, field; field = block.fieldRow[j]; j++) {
+        blockInfoList.push(field);
+      }
+
+      return this.infoBlocks[block.id];
+    },
+    inputType: function(connection) {
+      if (connection.check_) {
+        return connection.check_.join(', ').toUpperCase();
+      } else {
+        return 'any';
```

Comment:
I want upper case to signify types and lower case to signify colloquial words. Of course, this has very little significance to blind users since "all caps" isn't always read out to them.